From f1537152d336cd072fa4fe00ffe775e45d4df8d9 Mon Sep 17 00:00:00 2001
From: Mateusz <mateuszb@poczta.onet.pl>
Date: Sun, 4 Nov 2018 17:39:10 +0100
Subject: [PATCH 1/3] timeb.h: declare _ftime32 function, define _ftime
 according to _USE_32BIT_TIME_T

Signed-off-by: Mateusz Brzostek <mateuszb@poczta.onet.pl>
Signed-off-by: Liu Hao <lh_mouse@126.com>
(cherry picked from commit 38496499910a580ddc449a7d09f3bc0302767f31)
Signed-off-by: Jonathan Yong <10walls@gmail.com>
---
 mingw-w64-crt/lib-common/msvcrt.def.in      | 3 ++-
 mingw-w64-headers/crt/sec_api/sys/timeb_s.h | 4 +++-
 mingw-w64-headers/crt/sys/timeb.h           | 9 +++++----
 3 files changed, 10 insertions(+), 6 deletions(-)

diff --git a/mingw-w64-crt/lib-common/msvcrt.def.in b/mingw-w64-crt/lib-common/msvcrt.def.in
index 1a1db2b4..13115412 100644
--- a/mingw-w64-crt/lib-common/msvcrt.def.in
+++ b/mingw-w64-crt/lib-common/msvcrt.def.in
@@ -532,7 +532,8 @@ _fstat64
 _fstati64
 F64(_fstat64i32 == _fstat)
 _ftime
-F_ARM_ANY(_ftime32)
+F_I386(_ftime32 == _ftime)
+F_NON_I386(_ftime32)
 _ftime32_s
 _ftime64
 _ftime64_s
diff --git a/mingw-w64-headers/crt/sec_api/sys/timeb_s.h b/mingw-w64-headers/crt/sec_api/sys/timeb_s.h
index f635d32b..ed92946e 100644
--- a/mingw-w64-headers/crt/sec_api/sys/timeb_s.h
+++ b/mingw-w64-headers/crt/sec_api/sys/timeb_s.h
@@ -15,11 +15,13 @@ extern "C" {
 
 #if defined(MINGW_HAS_SECURE_API)
 
-  _CRTIMP errno_t __cdecl _ftime_s(struct __timeb32 *_Time);
+  _CRTIMP errno_t __cdecl _ftime32_s(struct __timeb32 *_Time);
   _CRTIMP errno_t __cdecl _ftime64_s(struct __timeb64 *_Time);
 
 #ifndef _USE_32BIT_TIME_T
 #define _ftime_s _ftime64_s
+#else
+#define _ftime_s _ftime32_s
 #endif
 #endif
 
diff --git a/mingw-w64-headers/crt/sys/timeb.h b/mingw-w64-headers/crt/sys/timeb.h
index c92c8e01..7bfe98e0 100644
--- a/mingw-w64-headers/crt/sys/timeb.h
+++ b/mingw-w64-headers/crt/sys/timeb.h
@@ -76,13 +76,14 @@ extern "C" {
 #endif
 
   _CRTIMP void __cdecl _ftime64(struct __timeb64 *_Time);
+  _CRTIMP void __cdecl _ftime32(struct __timeb32 *_Time);
 
-#ifdef _WIN64
+#ifndef _USE_32BIT_TIME_T
 #define _timeb __timeb64
-  _CRTIMP void __cdecl _ftime(struct __timeb64 *);
+#define _ftime _ftime64
 #else
 #define _timeb __timeb32
-  _CRTIMP void __cdecl _ftime(struct __timeb32 *);
+#define _ftime _ftime32
 #endif
 
 #ifndef _TIMESPEC_DEFINED
@@ -109,7 +110,7 @@ struct itimerspec {
   }
 #else
   __CRT_INLINE void __cdecl ftime(struct timeb *_Tmb) {
-    _ftime((struct __timeb32 *)_Tmb);
+    _ftime32((struct __timeb32 *)_Tmb);
   }
 #endif /* _USE_32BIT_TIME_T */
 #endif /* !__CRT__NO_INLINE */
-- 
2.14.1

